src/mlpack/tests/image_load_test.cpp

```diff
@@ -56,17 +60,121 @@ BOOST_AUTO_TEST_CASE(SaveImageAPITest)
   arma::Mat<unsigned char> im1;
   size_t dimension = info.Width() * info.Height() * info.Channels();
   im1 = arma::randi<arma::Mat<unsigned char>>(dimension, 1);
-  BOOST_REQUIRE(data::Save("APITest.bmp", im1, info, false, true) == true);
+  BOOST_REQUIRE(data::Save("APITest.bmp", im1, info, false) == true);
+
+  arma::Mat<unsigned char> im2;
+  BOOST_REQUIRE(data::Load("APITest.bmp", im2, info, false) == true);
+
+  BOOST_REQUIRE_EQUAL(im1.n_cols, im2.n_cols);
+  BOOST_REQUIRE_EQUAL(im1.n_rows, im2.n_rows);
+  for (size_t i = 0; i < im1.n_elem; ++i)
+    BOOST_REQUIRE_EQUAL(im1[i], im2[i]);
```

Comment:
Ahh, we should probably also add `remove("APITest.bmp")"` so that the file gets removed after the test is done.